bazel/testing.md

```diff
@@ -0,0 +1,74 @@
+# Testing With Bazel
+
+Bazel ships with support for executing tests and generating test coverage.
+
+## Execute Tests
+
+In general tests are executed by calling `bazel test` on one or multiple targets, where the targets are defined by a test rule.
+
+* Execute one test:
+`bazel test //lte/gateway/python/magma/mobilityd/tests:test_uplink_gw`
+
+* Execute all tests in folder:
+`bazel test //lte/gateway/python/magma/mobilityd/tests:all`
+Note: "all" is a keyword that expands all found test targets.
+
+* Execute all tests in lte:
+`bazel test //lte/...:all`
+
+* Execute all tests:
+`bazel test //...:all`
+
+* Why is my test not executed?
+  * A reason could be a test tagged as "manual", see below.
+
+## Helpful Flags
+
+For a detailed documentation see the bazel [command line reference](https://docs.bazel.build/versions/main/command-line-reference.html).
+
+| Flag | Effect |
+|-|-|
+| `--test_output=all` | print all test output |
+| `--sandbox_debug` | do not delete sandbox after test |
+| `--cache_test_results=no` | force rerun of tests, i.e., do not use cached results |
+
+## Execute Coverage
+
+In general coverage is generated by calling `bazel coverage` on one or multiple targets, where the targets are defined by a test rule. See "Execute Tests" for target expanding.
+
+Coverage data can be found in `$MAGMA_ROOT/bazel-out/_coverage/_coverage_report.dat`. For a quick preview you can generate a html report by, e.g.:
+`genhtml -o genhtml bazel-out/_coverage/_coverage_report.dat`
+The report can be opened via `$MAGMA_ROOT/genhtml/index.html`.
+
+## Tags
+
+Test targets can be enriched by tags, e.g.,
+```
+py_test(
+  ...
+  tags = [
+    "unittest",
+    "my_tag",
+  ]
+)
+```
+Tags can be used to filter tests to be executed:
+`bazel test --test_tag_filters=unittest,-my_tag ...`
+Execute all tests that are tagged with "unittest" but exclude all that are tagged with "my_tag".
+
+### Proposal For Tags Used In Magma
+
+ * All tests are tagged by their respective root folder, e.g., all tests in lte have the tag "**lte**".
+ * All tests are tagged by their respective module/service, e.g., all mobilityd tests are tagged by "**mobilityd**".
+ * All unit tests are tagged with "**unittest**".
+ * All "sudo tests" (Python specific tests that need to be executed as root) are tagged with "**sudo_test**" and "**manual**" (see below).
```

Comment:
I was thinking we can define all the special ones like `sudo_test` in a separate .bzl file (somewhere in the bazel/ directory) that other BUILD files can import from. In that case, we can better ensure we avoid typos and case mismatch. This would help us keep documentation updated if we ask people to propoerly document each new tag added in code reviews.